File: src/main/java/pl/exsio/querydsl/entityql/examples/jpa/entity/Order.java
package pl.exsio.querydsl.entityql.examples.jpa.entity;

import javax.persistence.*;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "ORDERS")
public class Order implements Serializable {

    @Id
    @Column(name = "ORDER_ID")
    private Long id;

    @OneToOne
    @JoinColumn(name = "USER_ID", nullable = false)
    private User user;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL)
    private List<OrderItem> items = new ArrayList<>();

}

File: kotlin-gradle-jpa-static/src/main/kotlin/pl/exsio/querydsl/entityql/examples/jpa/entity/Order.kt
package pl.exsio.querydsl.entityql.examples.jpa.entity


import java.io.Serializable
import javax.persistence.*

@Entity
@Table(name = "ORDERS")
class Order() : Serializable {

    @Id
    @Column(name = "ORDER_ID")
    @GeneratedValue
    var id: Long? = null

    @OneToOne
    @JoinColumn(name = "USER_ID", nullable = false)
    var user: User<Any>? = null

    @OneToMany(mappedBy = "order")
    var items: List<OrderItem>? = null

    @OneToMany(mappedBy = "orderReferenced")
    var itemsReferenced: List<OrderItem>? = null


}
